docs/build/apps/guestbook/frontend.mdx

```diff
@@ -0,0 +1,598 @@
+    // be used to reverse-lookup the smart wallet address, provided we know
+    // the passkey's ID (the user supplies that during the function's execution)
+    const { keyId_base64, contractId: cid } = await account.connectWallet({
+      getContractId,
+    });
+
+    // Store the key ID and contract address in our localStorage stores
+    keyId.set(keyId_base64);
+    console.log($keyId);
+    contractId.set(cid);
+    console.log($contractId);
+  } catch (err) {
+    console.log(err);
+    toastStore.trigger({
+      message: "Something went wrong logging in. Please try again later.",
+      background: "variant-filled-error",
+    });
+  }
+}
+```
+
+Similar, yet simpler, when compared with our `signup` function. We're using the `account.connectWallet` function. This function will:
+
+1. Trigger the user to authenticate, providing the passkey's ID along the way,
+2. Use Mercury to reverse-lookup the contract ID given that passkey ID, and finally
+3. Return the passkey ID and smart wallet address to our dapp.
+
+Great! Let's get the user logged out when they need to.
+
+#### User logout
+
+This is quite a bit easier than either signup or login functions. We don't really need to communicate with the Stellar network or Mercury here. All we'll do is clear out the user state, essentially.
+
+```js title="src/lib/components/connectButtons.svelte"
+async function logout() {
+  try {
+    // Reset the localStorage entry for the keyId
+    keyId.reset();
+    localStorage.removeItem("yog:keyId");
+
+    // Set the contract address store to an empty string
+    contractId.set("");
+
+    // Refresh the page, just for good measure
+    window.location.reload();
+  } catch (err) {
+    console.log(err);
+    toastStore.trigger({
+      message: "Something went wrong logging out. Please try again later.",
+      background: "variant-filled-error",
+    });
+  }
+}
+```
+
+With those three functions, our dapp is ready for users to authenticate with the dapp! Much easier than you probably expected it to be, right!?
+
+### The "profile menu"
+
+Still in our `connectButtons.svelte` component, we also have a collection of buttons and functions that represent a "profile menu" of sorts. The user can use these buttons to view their smart wallet balance, see it on Stellar Expert, send a donation to our (humble) guestbook maintainer, request more (Testnet) funding, etc. Much of this is unnecessary to dive into here in this tutorial, though I highly recommend taking a look at the [source code](https://github.com/ElliotFriend/ye-olde-guestbook/blob/main/src/lib/components/ConnectButtons.svelte) to get a better understanding of this functionality.
+
+However, we will look into the `donate` function here. This is a really useful example of how a dapp can enable their smart wallet users to interact with any asset on the Stellar network. (Here, we are using [Testnet] XLM for our asset, but the flow would be identical for _any_ asset you may want to use.)
```

Comment:
Is Testnet in brackets for a reason?

Reply:
I put it in brackets because it's inside a set of parenthesis (I think that's how i learned it in english class? 🤣). Either way, I shouldn't have it like that probably, b/c it's a) confusing, and b) might confuse the MDX parser which might expect a link.